### PR TITLE
PHP 8 Review: `LearningHistory`

### PR DESCRIPTION
**Pull request by mjansenDatabay**
Hello @alex40724, 

I completed the review of the `Services/LearningHistory` component.

Results:

- [x] The code style is `PSR-2 + X` compliant
- [x] No usage of `$_GET` / `$_POST` / `$_REQUEST` / `$_SESSION`
- [x] There is a `Unit Test Suite` with at least one unit test
- [x] The unit tests pass
- [x] External libraries are compatible with PHP 8 (if relevant)
- [ ] There are no serious `PhpStorm Code Inspection` issues left
- [ ] The types are fully documented (PhpDoc) or explict PHP types are used (type hints, return types, typed properties)

--

Actions needed:
- [ ] Please check my inline comments. Some of them are just information, some require an action. You can use `TODO PHP8-REVIEW` when searching for my comments.

Best regards,
@mjansenDatabay

### CODE REVIEW

File: Services/LearningHistory/classes/class.ilLearningHistoryService.php [hunk 1]
```diff
@@ -99,12 +99,14 @@ public function isActive(int $user_id = 0) : bool
             return false;
         }
 
-        if ($user_id = 0) {
+        if ($user_id === 0) {
```
Comment (mjansenDatabay) [inline Services/LearningHistory/classes/class.ilLearningHistoryService.php:102]: ;-)

File: Services/LearningHistory/classes/class.ilLearningHistoryProviderFactory.php [hunk 2]
```diff
@@ -47,7 +47,7 @@ public function getAllProviders(
     ) : array {
         $providers = array();
 
-        if ($user_id == 0) {
+        if ($user_id === 0 || $user_id === null) {
```
Comment (mjansenDatabay) [inline Services/LearningHistory/classes/class.ilLearningHistoryProviderFactory.php:50]: @alex40724 I was not 100% sure here

**Comment by alex40724** [comment]
Thanks, I will take care of the comments.

**Comment by alex40724** [comment]
Done.

**Pull request merged**